app/models/miq_group.rb

```diff
@@ -273,10 +273,19 @@ def current_user_group?
     id == current_user_group.try(:id)
   end
 
+  def current_group_for_any_user?
+    users.where(:current_group_id => id).count != 0
+  end
+
   def ensure_can_be_destroyed
     raise _("The login group cannot be deleted") if current_user_group?
     raise _("The group has users assigned that do not belong to any other group") if single_group_users?
     raise _("A tenant default group can not be deleted") if tenant_group? && referenced_by_tenant?
     raise _("A read only group cannot be deleted.") if system_group?
+    reset_current_group_for_users if current_group_for_any_user?
```

Comment:
This is a strange side effect of a method that seems to be checking if a group is allowed to be destroyed.

The way this is written it feels like this should be two separate steps:

- Ensure that the group in question is not the only group assigned to any user
- When the group is actually being removed, then reassign the current group

Reply:
> Ensure that the group in question is not the only group assigned to any user

Ah I see that this check is being done by the line above with `if single_group_users?`, but I still feel like we could add another hook for this, maybe `after_destroy`?

Reply:
I did consider that, however, the group should not be destroyed if the current_group cannot be changed for any user that has this as the current_group.

Reply:
As I read it the only reason the current_group would not be able to be changed is if a user had no other group to change to.  This case is already covered by another assertion already in this method, so I don't see any reason this can't be moved.

Reply:
I moved the changing of the current_group for the users that requires it to the around_destroy.